instrumentation/apache-camel-2.20/javaagent/src/test/groovy/io/opentelemetry/javaagent/instrumentation/apachecamel/SnsCamelTest.groovy

```diff
@@ -0,0 +1,696 @@
+            "net.transport" "IP.TCP"
+            "net.peer.port" {it == null || Number}
+          }
+        }
+      }
+      trace(3, 1) {
+
+        span(0) {
+          name "SNS.CreateTopic"
+          kind CLIENT
+          hasNoParent()
+          attributes {
+            "aws.agent" "java-aws-sdk"
+            "aws.endpoint" String
+            "aws.operation" "CreateTopic"
+            "aws.service" "AmazonSNS"
+            "http.flavor" "1.1"
+            "http.method" "POST"
+            "http.status_code" 200
+            "http.url" String
+            "net.peer.name" String
+            "net.transport" "IP.TCP"
+            "net.peer.port" {it == null || Number}
+          }
+        }
+      }
+      trace(4, 1) {
+
+        span(0) {
+          name "SNS.Subscribe"
+          kind CLIENT
+          hasNoParent()
+          attributes {
+            "aws.agent" "java-aws-sdk"
+            "aws.endpoint" String
+            "aws.operation" "Subscribe"
+            "aws.service" "AmazonSNS"
+            "http.flavor" "1.1"
+            "http.method" "POST"
+            "http.status_code" 200
+            "http.url" String
+            "net.peer.name" String
+            "net.transport" "IP.TCP"
+            "net.peer.port" {it == null || Number}
+          }
+        }
+      }
+      trace(5, 1) {
+        span(0) {
+          name "SQS.ReceiveMessage"
+          kind CLIENT
+          hasNoParent()
+          attributes {
+            "aws.agent" "java-aws-sdk"
+            "aws.endpoint" String
+            "aws.operation" "ReceiveMessage"
+            "aws.queue.url" queueUrl
+            "aws.service" "AmazonSQS"
+            "http.flavor" "1.1"
+            "http.method" "POST"
+            "http.status_code" 200
+            "http.url" String
+            "net.peer.name" String
+            "net.peer.port" {it == null || Number}
+            "net.transport" "IP.TCP"
+          }
+        }
+      }
+      trace(6, 5) {
+
+        span(0) {
+          name "input"
+          kind INTERNAL
+          hasNoParent()
+          attributes {
+            "apache-camel.uri" "direct://input"
+          }
+        }
+        span(1) {
+          name topicName
+          kind INTERNAL
+          childOf span(0)
+          attributes {
+            "apache-camel.uri" "aws-sns://${topicName}?amazonSNSClient=%23snsClient"
+            "messaging.destination" topicName
+          }
+        }
+        span(2) {
+          name "SNS.Publish"
+          kind CLIENT
+          childOf span(1)
+          attributes {
+            "aws.agent" "java-aws-sdk"
+            "aws.endpoint" String
+            "aws.operation" "Publish"
+            "aws.service" "AmazonSNS"
+            "http.flavor" "1.1"
+            "http.method" "POST"
+            "http.status_code" 200
+            "http.url" String
+            "net.peer.name" String
+            "net.peer.port" {it == null || Number}
+            "net.transport" "IP.TCP"
+          }
```

Comment:
Most of those AWS spans are almost the same, they only differ in names, kinds and `aws.operation` attributes - can you add a private method that encapsulates all other properties/attributes? This'll make those tests much shorter and easier to read

Reply:
Yeah, I plan to clean-up all of Camel / AWS propagation tests once I'm done (I still want to test two additional scenarios). So another PR will include this 👍